Working prototype for indenting entire function on .

onno-vos-dev opened the pull request:
Co-authored-by: Andreas Hasselberg <andreas.hasselberg@gmail.com>

Rebased to try and get a green build, let's see how it goes :)

onno-vos-dev committed:
Working prototype for indenting entire function on .

Co-authored-by: Andreas Hasselberg <andreas.hasselberg@gmail.com>

Delete iostr.erl

Fix linter complaints

Fix proper error

Fix typespec

Maybe make Dialyzer happy?

Temp fixes to get dialyzer to run

Add spec

Better fix for dialyzer errors

Add config to enable on type formatting, defaulting to false

Skip formatting if . is on a commented out line

onno-vos-dev commented:
> I think we should consider \t for the prototype and then, when it is configurable switch the defaults to ".", ";" and "\t".

I tried but it seems that VsCode transforms \t to spaces on the fly so the \t never lands on ErlangLS. I'd imagine a similar thing happens for other editors...

onno-vos-dev commented:
> This is no longer used, right?

Removed :+1:

onno-vos-dev commented:
> I think we lost the code that removes the tmp file.

tempfile takes care of that for us :) It executes the fun and than removes the tempdir it created :)

onno-vos-dev commented:
> I'm unsure if this is used.

Neither am I, I'll give it a try later

andreashasse commented:
> > I think we lost the code that removes the tmp file.
> 
> tempfile takes care of that for us :) It executes the fun and than removes the tempdir it created :)

True, I read the code incorrectly :)

andreashasse commented on apps/els_lsp/src/els_formatting_provider.erl:
```diff
@@ -133,5 +136,70 @@ rangeformat_document(_Uri, _Document, _Range, _Options) ->
+          %% than jumping to a newline
+          NewText =
+            string:trim(
+              tempdir:mktmp(ParseF), trailing, "\n"),
```
Nice

onno-vos-dev commented:
- [ ] Add tests
- [ ] Try to figure out \t instead of . to terminate on.
- [ ] Disable OnRangeFormatting by default, enable to get the behaviour introduced in this PR

andreashasse commented on apps/els_core/src/els_client.erl:
```diff
@@ -431,6 +453,10 @@ request_params({initialize, {RootUri, InitOptions}}) ->
                   , <<"hover">> =>
                     #{ <<"contentFormat">> => ContentFormat }
+                  , <<"documentOnTypeFormattingProvider">> =>
+                      #{ firstTriggerCharacter => <<",">>
```
I find it weird that the keys are atoms and the value is `<<,>>`, not `<<.>>`.

onno-vos-dev replied:
I know but proper apparently complained when they were binaries...

garazdawi committed:
Add gradualizer diagnostic

garazdawi committed:
Make it display an error

garazdawi committed:
First working example

garazdawi committed:
Add better error reporting

garazdawi committed:
Integrate gradualizer with pois

garazdawi committed:
Disable column number check again

garazdawi committed:
Fix compiler errors

garazdawi committed:
Fix loading of things into db

erszcz committed:
Pass includes to Gradualizer

garazdawi committed:
Refactor and try again

erszcz committed:
Make bound_var_in_pattern test filter the actual diagnostics

garazdawi committed:
Use gradualizer_db:import_erl_files/2 if available

erszcz committed:
wip: Add a test for Gradualizer diagnostics

garazdawi committed:
Fix start_and_load

plux committed:
Add support for renaming types

plux committed:
Implement incremental sync

gomoripeti committed:
Support record fields in reference provider

References are only searched in the document containing the definition and
documents that (maybe recursively) include the defining document.

The scoping for macro and record referencing is also changed to the same.

As a sideeffect macros and records are not stored in the els_dt_references table
any more.

plux committed:
Allow renaming function when pointing at function application

plux committed:
Add support for renaming from export_entry and import_entry tools

Refactor tests

robertoaloi committed:
Merge pull request #1004 from plux/rename-function-from-reference

Allow renaming function when pointing at function reference

robertoaloi committed:
Improve installation instructions.

robertoaloi committed:
Merge pull request #1003 from pierre-rouleau/main

Show how to install erlang_ls

robertoaloi committed:
Merge pull request #1000 from hellmean/feature/truncate_spec

[FEATURE] truncate suggested spec title

plux committed:
Properly extract line number from dialyzer warning

Seems like this was broken in OTP 24 as the warning now returns a line and
column. Use erl_anno:line/1 which will handle both Line and {Line, Col}

robertoaloi committed:
Merge pull request #1006 from plux/fix-dialyzer-diagnostics-crash

Properly extract line number from dialyzer warning

plux committed:
Make more exact dialyzer diagnostics by utilizing column number

plux committed:
rebuild

plux committed:
Bump timeout in els_server_SUITE to avoid CI fails

robertoaloi committed:
Merge pull request #1007 from plux/improve-dialyzer-diagnostic-range

Make more exact dialyzer diagnostics by utilizing column number

robertoaloi committed:
Merge pull request #1001 from gomoripeti/record_field_refs

Support record fields in reference provider

michalmuskala committed:
Fix navigation to macros with overloaded arity

Macro name should be identified by both name & arity (if available).
Macros can be overloaded on arity. Currently, if that happens, the
navigation is incorrect and points to the first definition with
particular name, regardless of the actual arity.

plux committed:
Upgrade ranch dep to 2.0.0

plux committed:
Fix find references for included types

robertoaloi committed:
Merge pull request #1009 from plux/ranch-2.0

Upgrade ranch dep to 2.0.0

robertoaloi committed:
Merge pull request #1010 from plux/fix-included-type-references

Fix find references for included types

michalmuskala committed:
Add snippets for macro completions

robertoaloi committed:
Merge pull request #1002 from plux/incremental-sync

Implement support for incremental sync

elbrujohalcon committed:
Allow config files to be called erlang_ls.yml

elbrujohalcon committed:
Use yaml instead of yml

robertoaloi committed:
Merge pull request #1014 from elbrujohalcon/erlang_ls.yml

Allow config files to be called erlang_ls.yaml

misaki214 committed:
Fix file path on Windows

gomoripeti committed:
Support renaming records and record fields

zsoci committed:
Add support for using long names in dap for project node

robertoaloi committed:
Merge pull request #1020 from gomoripeti/rename_record

Support renaming records and record fields

robertoaloi committed:
Merge pull request #1017 from misaki214/windows

Fix file path on Windows

RunyaoZhang committed:
[dap] support "attach" (#1015)

* [dap] support "attach"

Different from "launch" which will start a shell, "attach" will attach to an existing shell.
See https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Attach

* [dap] support "attach" 2

Add continue and stop interpreting.

* [dap] support "attach" 3

Add dap rpc int:n/1 and int: interpreted/0

* [dap] support "attach" 4

Fix lint error

keynslug committed:
Prevent infinite recursion when enumerating document POIs

Before this commit recursion in `els_scope:included_pois/2` might
become infinite. This can happen for a couple of reasons:
* header mistakenly includes itself,
* header ends up including itself transitively, though inclusion
  is guarded with `ifndef` and `define` to prevent compiler from
  complaining.
* header includes another header with exactly same filename but
  located in a different directory, yet from the point of view of
  `els_utils:find_header/1` it's the same header.

This commit introduces simple abstract functional graph concept
with traversals over set of nodes reachable from a given node
and employs it to:
* gather all POIs reachable through inclusion in the `els_scope`
  module,
* construct digraph used in 'unused includes' diagnostics.

TheGeorge committed:
[dap] dummy handler for pause

robertoaloi committed:
[#1027] Remove TCP support

robertoaloi committed:
Merge pull request #1028 from erlang-ls/1027-remove-tcp-support

[#1027] Remove TCP support

robertoaloi committed:
Merge pull request #1026 from TheGeorge/dap_pause

[dap] dummy handler for pause

gomoripeti committed:
Fix docs for macro with arguments

Fixes #1022

robertoaloi committed:
Merge pull request #1019 from zsoci/uselongnames

Add support for using long names in dap for project node

robertoaloi committed:
Merge pull request #967 from plux/rename-type

Add support for renaming types

gomoripeti committed:
Fix regression: completion for incomplete export enrty

The unexpected atom (an incomplete function name) in the export attribute caused
the below crash in the parser. This way the completion provider failed to
determine that the location is in an export context, and used function snippet
with arguments instead of '/arity'.

For example when hitting enter on a completion erlang_ls used to insert
`foo(Arg)` instead of `foo/1`.

```
1> els_parser:parse("-export([foo]).").
2021-06-06T16:40:30.522244+02:00 warning: Please report error parsing form error:{try_clause,foo}:[{els_parser,'-skip_function_entries/1-fun-0-',1,[{file,"/Users/ext.pgomori/git/erlang_ls/apps/els_lsp/src/els_parser.erl"},{line,839}]},{lists,'-filter/2-lc$^0/1-0-',2,[{file,"lists.erl"},{line,1286}]},{els_parser,attribute_subtrees,2,[{file,"/Users/ext.pgomori/git/erlang_ls/apps/els_lsp/src/els_parser.erl"},{line,792}]},{els_parser,fold,3,[{file,"/Users/ext.pgomori/git/erlang_ls/apps/els_lsp/src/els_parser.erl"},{line,687}]},{els_parser,'-parse_forms/1-lc$^0/1-0-',1,[{file,"/Users/ext.pgomori/git/erlang_ls/apps/els_lsp/src/els_parser.erl"},{line,59}]},{els_parser,parse,1,[{file,"/Users/ext.pgomori/git/erlang_ls/apps/els_lsp/src/els_parser.erl"},{line,30}]},{erl_eval,do_apply,6,[{file,"erl_eval.erl"},{line,684}]},{shell,exprs,7,[{file,"shell.erl"},{line,686}]}], {attribute,#{end_location => {1,16},location => {1,1}},{atom,#{end_location => {1,8},location => {1,2},text => "export"},export},[{list,#{end_location => {1,14},location => {1,9}},[{atom,#{end_location => {1,13},location => {1,10},text => "foo"},foo}]}]}
```

robertoaloi committed:
Merge pull request #1029 from gomoripeti/marco_edoc

Fix docs for macro with arguments

robertoaloi committed:
[#1031] Find implementations for callback functions

robertoaloi committed:
Merge pull request #1030 from gomoripeti/export_entry_completion

Fix regression: completion for incomplete export enrty

TheGeorge committed:
[els_core] add logging via client notifications

robertoaloi committed:
Merge pull request #1032 from erlang-ls/1031-find-implementations-for-callback-functions

[#1031] Find implementations for callback functions

robertoaloi committed:
Merge pull request #1033 from TheGeorge/client_logging

add logging via client notifications

robertoaloi committed:
[#992] Cleanup logging (#1034)

* Use timestamp as first entry
* Move message to the front to improve readibility (left-alignment for
messages)
* Remove un-necessary new lines
* Rename ELP -> LSP
* Remove redundant pid from Background job log messages
* Use macros instead of numeric literals for message types
* Shorten mapping function name

gomoripeti committed:
Fix creating macro POIs in function applications

robertoaloi committed:
Merge pull request #1036 from gomoripeti/macro_in_application

Fix creating macro POIs in function applications

robertoaloi committed:
Merge pull request #1023 from keynslug/fix/scope-include-loop

Prevent infinite recursion when enumerating document POIs

ztion committed:
Add support for hover on record expression.

Commit:
Update BSP support

Works with https://github.com/erlang-ls/rebar3_bsp/pull/9/commits

Updates BSP support to include fetching app & dependency dirs and cleans up overall structure of the implementation.

zsoci committed:
Refactor attach and launch

zsoci committed:
Make start_distribution type aware

pierre-rouleau committed:
Ignore Emacs backup and temporary files.

pierre-rouleau committed:
...With a terminating newline.

robertoaloi committed:
Merge pull request #1038 from ztion/hover_record

Add support for hover on record expression.

plux committed:
Fix crash when applying edits to unicode text

keynslug committed:
Improve unused includes diagnostic runtime

...By considering only those POIs relevant for this diagnostic, e.g.
POIs which MAY potentially be defined in headers. This often
significantly cuts number of POIs to iterate over which
directly speed up diagnostic.

zsoci committed:
Revert rebar.lock

robertoaloi committed:
Merge pull request #1042 from plux/fix-incremental-sync-unicode-issue

Fix crash when applying edits to unicode text

robertoaloi committed:
Merge pull request #1041 from zsoci/testlongnames

Testlongnames

robertoaloi committed:
Merge pull request #1040 from pierre-rouleau/main

Ignore Emacs backup and temporary files

robertoaloi committed:
[#1052] Add support for ?MODULE macro as record name

robertoaloi committed:
Merge pull request #1053 from erlang-ls/1052-handle-module-macro-as-record-name

[#1052] Add support for ?MODULE macro as record name

robertoaloi committed:
[#1049] Lower log level for background jobs and compile info/options to DEBUG

robertoaloi committed:
[#1052] Do not crash if macro name is not MODULE

robertoaloi committed:
Merge pull request #1055 from erlang-ls/1052-relax-matching-for-macro-names

[#1052] Do not crash if macro name is not MODULE

robertoaloi committed:
Merge pull request #1054 from erlang-ls/1049-lower-level-for-some-debug-messages

[#1049] Lower log level for background jobs and compile info/options to DEBUG

robertoaloi committed:
[#990] Show warning to the user in case of missing config file

robertoaloi committed:
Merge pull request #1056 from erlang-ls/990-diagnostics-for-config-file

[#990] Show warning to the user in case of missing config file

robertoaloi committed:
Detect flat lists of printable characters in logpoints and output them as strings

robertoaloi committed:
Merge pull request #1059 from erlang-ls/improve-printing-of-logpoints

Detect flat lists of printable characters in logpoints and output them as strings

robertoaloi committed:
[#1060] Quick fix for the DAP regression

A more appropriate solution, removing the dependency on the els_config module, will be provided separately

robertoaloi committed:
Merge pull request #1061 from erlang-ls/fix-dap-regression

[#1060] Quick fix for the DAP regression

robertoaloi committed:
Merge pull request #1043 from keynslug/perf/faster-unused-includes

Improve unused includes diagnostic runtime

robertoaloi committed:
Specify category for DAP events

robertoaloi committed:
Merge pull request #1062 from erlang-ls/specify-category-for-dap-events

Specify category for DAP events

robertoaloi committed:
[#240] Publish escripts

robertoaloi committed:
Merge pull request #1063 from erlang-ls/240-publish-escripts

[#240] Publish escripts

robertoaloi committed:
[#625] Add guidelines on encoding strings

robertoaloi committed:
Merge pull request #1065 from erlang-ls/625-add-guidelines-on-encoding-binaries

[#625] Add guidelines on encoding strings

TheGeorge committed:
[dap] fix continue request response

The continue requests reponse defines an optional `allThreadsContinued` field
that defaults to true. This breaks the DAP when debugging multiple threads

robertoaloi committed:
[#1070] Only set the cookie if the provided one is a valid one

This fixes a recently introduced regression. The `erlang:get_cookie/0`
was moved earlier, to a stage where distribution was not started
yet. Since that function returns `nocookie` in a system with no
distribution, that caused the atom `nocookie` to be erroneously used
as a cookie. A cleaner approach could be used, but that would require
a bit of refactoring. For now this should fix the regression with
minimal code changes.

robertoaloi committed:
Merge pull request #1069 from TheGeorge/fix_dap_continue_response

[dap] fix continue request response

robertoaloi committed:
Merge pull request #1071 from erlang-ls/1070-fix-distribution-when-no-cookie-is-specified

[#1070] Only set the cookie if the provided one is a valid one

NAR committed:
Add config option to exclude unused headers from warnings.

A new option is added to the configuration (exclude_unused_includes).
Header files specified by this configuration option are not checked if
they are used or not.

NAR committed:
[#1073] Find non-parametrized macro definitions.

Some macro definitions that accept parameters do not actually specify
them at declaration, so erlang_ls doesn't find them. One such example is
the ?config macro in the common_test ct.hrl file. So in case a macro
usage looks like it passes some parameters, but erlang_ls can't find a
macro definition that accepts parameters, let's try to find a macro
definition without a parameter list (based on experiments the erlang
compiler seems to do this too).

NAR committed:
[#1073] Add test to find non-parametrized macro definitions.

The code calls ?MACRO_B/2, but has only a ?MACRO_B definition without
parameters. As this actually works, erlang_ls should find the
definition, so let's test this.

The test changes the code_navigation.erl file, so a few other tests had
to be updated also to cope with this change.

Commit:
Fix flaky test purge_references

Do not drop the indexingEnabled InitOption.

Commit:
Use gen_server send request wrappers also on OTP 23

Turns out gen_server:receive_response was added in OTP 24, so use
the wrappers also on OTP 23.

Commit:
Stop polluting log with error prints about missing BSP config

This is not an unusual case, and not having a config isn't an error.

robertoaloi committed:
Merge pull request #1072 from NAR/exclude-unused-imports

Add config option to exclude unused headers from warnings.

NAR committed:
Break too long line into multiple lines.

Commit:
Fix badarg in case of unused macro with parameters.

robertoaloi committed:
Merge pull request #1048 from al-khanji/main

BSP support

robertoaloi committed:
Merge pull request #1074 from NAR/fix-finding-macro-definition

[#1073] Find non-parametrized macro definitions.

Commit:
Simplify protocol implementation

Use erlang:decode_packet/3 to do the heavy lifting for us. Less code,
and might also be faster.

alanz committed:
Introduce diagnostic codes for the OTP compile

Expose the module and error passed to Module:print_error as an error code.

This allows us to provide telemetry of the codes, for later analysis, in
situations where this is practical.

It also allows an easy tie-up to any external documentation about how to deal
with the error.

The mechanism is to use the source of the original format_error/2 clauses to map
onto codes, which have a regular format, e.g. L1227 where the first letter
indicates the linter, and the rest is just a 4 digit unique number.

This format fits in with with existing error presentation in IDEs, as it is
already used by other compilers such as rust.

robertoaloi committed:
Merge pull request #1076 from nwalker/unused-macro-type-mismatch

Fix type mismatch in unused_macros diagnostic.

robertoaloi committed:
Enable incremental_sync by default

robertoaloi committed:
Merge pull request #1081 from erlang-ls/enable-incremental-sync-by-default

Enable incremental_sync by default

robertoaloi committed:
Merge pull request #1078 from erlang-ls/diagnostics-codes

Introduce diagnostic codes for the OTP compile

robertoaloi committed:
Merge pull request #1077 from al-khanji/main

Simplify protocol implementation

alanz committed:
Send telemetry notification with compiler diagnostic codes

When enabled, default off, send a `telemetry/event` notification containing the
URI and error codes for compiler diagnostics in the given file.

The allows a client to log the telemetry, and perform later analysis on the
occurrence of diagnostics types.

Note: any logging takes place at the client level, if enabled, to whatever
location makes sense for the specific client.

robertoaloi committed:
Merge pull request #1079 from erlang-ls/diagnostics-codes-telemetry

Send telemetry notification with compiler diagnostic codes

maxnordlund committed:
Fix DAP not finding the source for a module

This makes the els_dap_rpc try much harder to find the source, by first
asking `int:file/1`, then `code` and `filelib:find_source/1` and
finally falling back to the current behaviour of using `module_info`
and checking the `source` compilation option.

This allows you to build dependencies/OTP in one place, and run it from
another.

robertoaloi committed:
Merge pull request #1084 from maxno-kivra/mn-fix-dap-find-source

Fix DAP not finding the source for a module

robertoaloi committed:
[#1047] Mark headers with compiler attributes as used

robertoaloi committed:
[#1086] Temporarily skip coveralls send

robertoaloi committed:
Merge pull request #1087 from erlang-ls/1086-skip-coveralls-until-root-cause-is-found

[#1086] Temporarily skip coveralls send

robertoaloi committed:
Merge pull request #1085 from erlang-ls/1047-mark-headers-with-compiler-attribute-as-used

[#1047] Mark headers with compiler attributes as used

robertoaloi committed:
[#1067] Remove unused File parameter

robertoaloi committed:
[#1067] Handle elvis.config errors

robertoaloi committed:
Merge pull request #1088 from erlang-ls/1067-handle-elvis-config-errors

[#1067] Handle Elvis config errors

gomoripeti committed:
Hover docs for fun expressions referring to local or remote functions

erszcz committed:
Don't cover up Gradualizer startup failures

erszcz committed:
Don't make the Gradualizer diagnostic a default one

robertoaloi committed:
[#1094] Add support for Call Hierarchy

TheGeorge committed:
user background job for hover provider to avoid blocking request queue on large documents

TheGeorge committed:
move db operations of hover provider into background job

TheGeorge committed:
fix dialyzer warnings in els_hover_provider

robertoaloi committed:
Merge pull request #1091 from TheGeorge/async_hover

user background job for hover provider to avoid blocking request queue

robertoaloi committed:
Merge pull request #1096 from erlang-ls/1094-add-support-for-call-hierarchy

[#1094] Add support for Call Hierarchy

gomoripeti committed:
Update erlfmt to 1.0.0

robertoaloi committed:
[#1098] Detect unused record fields

robertoaloi committed:
[#1098] More recordy name, usort list, no dead code

robertoaloi committed:
Merge pull request #1099 from erlang-ls/1098-detect-unused-record-fields

[#1098] Detect unused record fields

garazdawi committed:
Vendor Erlang/OTP shell_docs rendering

garazdawi committed:
Implement markdown rendering

gomoripeti committed:
Refactor record poi parsing

Change of behaviour:
- The record POI now has only a list of atoms in data -> field_list.
  The default value syntax trees were not used and some memory can be saved this
  way.
- ?MODULE macro as record name also handled for record type (eg
  go-to-definintion also works from types and specs)

gomoripeti committed:
Move macro related parser tests to separate suite

gomoripeti committed:
Add more macro parsing tests

robertoaloi committed:
[#1107] Shutdown request does not include parameters

robertoaloi committed:
Merge pull request #1108 from erlang-ls/1107-do-not-always-expect-params

[#1107] Shutdown request does not include parameters

garazdawi committed:
Add lookup of docs for types and use edoc chunks

garazdawi committed:
Add eep48 to elvis ignore list

garazdawi committed:
Fix compilation before Erlang/OTP 23

garazdawi committed:
Fix dialyzer error

dgud committed:
[#859] Ignore test data directories.

Test data directories may contain erlang files with modules names
that clashes with application or system modules, ignore these
directories.

As one example in otp "lib/dialyzer/test/r9c_SUITE_data/src/" contains
the many files that clashes. This causes among other things that
jump to definition almost always jumps to the wrong place when working
inside OTP.

garazdawi committed:
Fix a bunch of doc rendering issues

garazdawi committed:
Fix Erlang/OTP < 23 dialyzer issue

robertoaloi committed:
Merge pull request #1109 from dgud/dgud/filter-test-dirs

[#859] Ignore test data directories.

robertoaloi committed:
[#1110] Discover test files automatically

robertoaloi committed:
Merge pull request #1111 from erlang-ls/1110-discover-test-files-automatically

[#1110] Discover test files automatically

garazdawi committed:
Introduce local group_leader for doc to capture output

garazdawi committed:
Add els_utils:find_modules for fetching all variants of a module

When there are multiple versions of the same file we sometimes
want to check all of them.

garazdawi committed:
Build edoc doc-chunks in user cache directory

garazdawi committed:
Check all variants of a module for docs before generating

When developing Erlang/OTP all the modules (i.e. lists.erl)
show up at least twice so when looking for the EEP-48 style
docs we should check all of them in prioritization order for
docs and display the first we find.

garazdawi committed:
Remove dead markup

Bold is not rendered within a pre

garazdawi committed:
Polish fetching of type documentation

garazdawi committed:
Do not include private functions when generating edoc

garazdawi committed:
Change -type poi location and create poi for builtin types

This makes the -type poi act more like -spec, which is needed
for fetching the -type for rendering docs.

The builtin types have documentation in 24.2, so we should create
pois for them so that we can use that in future versions.

garazdawi committed:
Do not generate hr if no other docs are to be shown

garazdawi committed:
Fix testcases

gomoripeti committed:
Make handling macros in record name position consistent

The subtrees function for some node types always skipped record name element,
for other types it always included it. This resulted in the inconsistency that
sometimes no macro POI was created, in other cases a macro POI was created for
?MODULE macro as well (which would prevent eg. go to definition of the record).

robertoaloi committed:
Merge pull request #1112 from gomoripeti/hover_implicit_fun

Hover docs for fun expressions referring to local or remote functions

JimMoen committed:
Add make target `clean` to quick rebuild

robertoaloi committed:
Merge pull request #1113 from gomoripeti/update_erlfmt

Update erlfmt to 1.0.0

robertoaloi committed:
Merge pull request #1116 from JimMoen/more-make-target

Add make target `clean`

erszcz committed:
Extract FmtOpts

erszcz committed:
Format els_gradualizer_diagnostics according to .editorconfig

erszcz committed:
Capture multiline Gradualizer messages (use re 'dotall')

erszcz committed:
Depend on Gradualizer

erszcz committed:
Finish the Gradualizer diagnostics test case

garazdawi committed:
Include type docs in completion

garazdawi committed:
Fix crash when macro definition cannot be found

garazdawi committed:
Indentation fix

garazdawi committed:
Remove dead code from eep48 module

garazdawi committed:
Do not render docs for functions/types with none docs

The docs will be create by els_docs instead.

garazdawi committed:
Fix completion testcases

erszcz committed:
Update Gradualizer (fixed build time warnings)

erszcz committed:
Apply post-review fixes

garazdawi committed:
Fix testcases

robertoaloi committed:
Merge pull request #1117 from erszcz/gradualizer

Provide a Gradualizer diagnostic

garazdawi committed:
Add specs to all els_eep48 functions

garazdawi committed:
Fix rename of opaque types

garazdawi committed:
Add rename_opaque testcase

garazdawi committed:
Add testing of edoc completion docs

robertoaloi committed:
[#1128] Refactor test framework for diagnostics

garazdawi committed:
Fix sending of unicode text in els_client

garazdawi committed:
Expand testing of documentation for completion items

garazdawi committed:
Added test for hover documentation

garazdawi committed:
Fix line length issues

robertoaloi committed:
Merge pull request #1100 from garazdawi/eep-48-markdown

Eep 48 markdown

dgud committed:
Do not recurse in els_utils:fold_files

Avoids double recursion, recursion is done in els_config.

gomoripeti committed:
Remove fragile heuristics from editable_range

robertoaloi committed:
Merge pull request #1126 from gomoripeti/editable_range

Remove fragile heuristics from editable_range

robertoaloi committed:
Merge pull request #1122 from dgud/dgud/filter-test-dirs

Do not recurse in els_utils:fold_files

robertoaloi committed:
Merge pull request #1129 from erlang-ls/1128-simplify-diagnostics-tests

[#1128] Refactor test framework for diagnostics

pablocostass committed:
Bring back `coveralls send` step in CI

pablocostass committed:
Update coverage badge with proper default branch

robertoaloi committed:
Merge pull request #1130 from pablocostass/1086-bring-coveralls-back

Bring back `coveralls send` step in CI

adrianroe committed:
Remove TCP support

The commit in
https://github.com/erlang-ls/erlang_ls/commit/d4a38eeb7317887c7860ffd43920219f22c2a969
removed TCP support but left some references to ranch which causes
issues in a nix build environment.

robertoaloi committed:
Merge pull request #1131 from id3as/remove_ranch

Remove TCP support

robertoaloi committed:
[1140] Do not include error description in code

robertoaloi committed:
Merge pull request #1141 from erlang-ls/1140-do-not-assume-string-descriptions-in-make-code

[1140] Do not include error description in code

robertoaloi committed:
[#1143] Lower docs log message level to DEBUG

Fixes #1142, #1143.

robertoaloi committed:
Merge pull request #1144 from erlang-ls/1143-lower-edoc-log-messages-to-debug

[#1143] Lower docs log message level to DEBUG

gomoripeti committed:
Parse incomplete text

In case of unparsable forms, try removing some trailing tokens, this way the
first half of the form could be parsed and POIs extracted.

related to #1037

garazdawi committed:
Fix unicode output via stdout

garazdawi committed:
Fix definition list to work with inline code

We change indent to be " " intead of nbsp as that seems
to be what vscode markdown expects for bullets.

robertoaloi committed:
Merge pull request #1147 from garazdawi/lukas/fix-unicode-os-cmd

Fix unicode output via stdout

robertoaloi committed:
[#1149] Exit with 0 on --version

robertoaloi committed:
Merge pull request #1150 from erlang-ls/1149-version-should-exit-0

[#1149] Exit with 0 on --version

garazdawi committed:
Revert unicode support

The support to render unicode broke the latin1 support
so until we fix both we fall back to latin1 so that we
can open such files...

People really should update their files to be unicode!

robertoaloi committed:
Merge pull request #1151 from garazdawi/lukas/revert-unicode-support

Revert unicode support

robertoaloi committed:
[#1152] Emit error in case the module name does not coincide with the filename

robertoaloi committed:
Merge pull request #1156 from erlang-ls/1152-emit-module-name-error

[#1152] Emit error in case the module name does not coincide with the filename

robertoaloi committed:
Merge pull request #1146 from gomoripeti/incomplete

Parse incomplete text

Commit:
[#1050] Filter log notification method to avoid recursion

robertoaloi committed:
Precompute list of enabled diagnostics

gomoripeti committed:
Fix parsing trailing comments in modules

erlfmt_parser returns a `raw_string` form for comments which are after the last
form in the file and not followed by any more form. It was not expected by
els_parser that a raw_string can contain no tokens (just comments and
whitespace). A file with no code just whitespace has the same result.

Fixes #1171

robertoaloi committed:
Merge pull request #1158 from AminArria/fix-recursive-logs

[#1050] Add separate way to send log notification to avoid recursion

lukebakken committed:
Take CRLF into account

Related to https://github.com/inaka/elvis_core/issues/220

robertoaloi committed:
Merge pull request #1172 from erlang-ls/precompute-enabled-diagnostics

Precompute list of enabled diagnostics

kianmeng committed:
Fix typos

robertoaloi committed:
[#1174] [emacs] Ensure keybinding is configured before loading lsp-mode

lukebakken committed:
Update elvis_core

robertoaloi committed:
Merge pull request #1175 from erlang-ls/1174-set-keymap-prefix-after-loading-lsp-mode

[#1174] [emacs] Ensure keybinding is configured before loading lsp-mode

robertoaloi committed:
Merge pull request #1178 from gomoripeti/parse_trailing_comment

Fix parsing trailing comments in modules

robertoaloi committed:
Merge pull request #1176 from lukebakken/crlf-line-endings

Update elvis_core to deal with CRLF line endings

robertoaloi committed:
Merge pull request #1173 from kianmeng/fix-typos

Fix typos

plux committed:
Augment local config onto global config

robertoaloi committed:
Merge pull request #1184 from plux/augment-global-config

Augment local config onto global config

plux committed:
Fix issues with rename variable involving specs (#1183)

sirikid committed:
Completion without snippets (#1170)

plux committed:
Ensure that default node name is valid (#1182)

This fixes an issue where initialization of els distribution would fail
due to invalid node name if the project directory contains characters
that are not valid in an erlang node name.

plux committed:
Fix parsing of define arguments (#1186)

HiPhish committed:
Allow installing into custom directory (#1188)

plux committed:
Support renaming function when standing on spec (#1190)

plux committed:
Improve goto definition for variables (#1187)

* Use same scoping rules as variable renaming
* No longer limited to only variables inside functions
* Add support for finding variable references

robertoaloi committed:
Avoid compiler_diagnostics crash in case module name is invalid (#1191)

Invalid syntax on the module attribute can cause a crash.

onno-vos-dev committed:
Maybe fix formatting?
